session-replay/Sources/DatadogSessionReplay/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UINavigationBarRecorder.swift

```diff
@@ -0,0 +1,73 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import UIKit
+
+internal struct UINavigationBarRecorder: NodeRecorder {
+    func semantics(of view: UIView, with attributes: ViewAttributes, in context: ViewTreeSnapshotBuilder.Context) -> NodeSemantics? {
+        guard let navigationBar = view as? UINavigationBar else {
+            return nil
+        }
+
+        let builder = UINavigationBarWireframesBuilder(
+            wireframeRect: inferOccupiedFrame(of: navigationBar, in: context),
+            wireframeID: context.ids.nodeID(for: navigationBar),
+            attributes: attributes,
+            color: inferColor(of: navigationBar)
+        )
+
+        return SpecificElement(wireframesBuilder: builder, recordSubtree: true)
+    }
+
+    private func inferOccupiedFrame(of navigationBar: UINavigationBar, in context: ViewTreeSnapshotBuilder.Context) -> CGRect {
+        // TODO: RUMM-2791 Enhance appearance of `UITabBar` and `UINavigationBar` in SR
+        var occupiedFrame = navigationBar.frame
+        for subview in navigationBar.subviews {
+            let subviewFrame = subview.convert(subview.bounds, to: context.coordinateSpace)
+            occupiedFrame = occupiedFrame.union(subviewFrame)
+        }
+        return occupiedFrame
+    }
+
+    private func inferColor(of navigationBar: UINavigationBar) -> CGColor {
+        // TODO: RUMM-2791 Enhance appearance of `UITabBar` and `UINavigationBar` in SR
+        if #available(iOS 13.0, *) {
+            switch UITraitCollection.current.userInterfaceStyle {
+            case .light:
+                return UIColor.white.cgColor
+            case .dark:
+                return UIColor.black.cgColor
+            default:
+                return UIColor.white.cgColor
+            }
+        } else {
+            return UIColor.white.cgColor
+        }
+    }
+}
```

Comment:
Getting the actual color of `UINavigationBar` (same for `UITabBar`) is not easy. Neither the `UINavigationBar` view, nor any of its subviews seems to have this information set explicitly:

<img width="1394" alt="Screenshot 2022-12-07 at 20 04 00" src="https://user-images.githubusercontent.com/2358722/206272613-8934f208-b525-47a9-8b97-ebb7cd18687e.png">

I suppose, the actual color is achieved with [Core Image filters](https://developer.apple.com/documentation/quartzcore/calayer/1410901-filters) configured on `_UIVisualEffectBackdropView` child. Notice "color saturate" on this list:

<img width="887" alt="Screenshot 2022-12-07 at 20 00 37" src="https://user-images.githubusercontent.com/2358722/206272964-b74fa304-d98a-4b9e-a193-cdf2af3d0b33.png">

In any case, this is not quick win, so I created separate task to address it.